apstra/api_iba_widgets.go

```diff
@@ -17,20 +17,20 @@ const (
 type IbaWidgetType enum.Member[string]
 
 var (
-	IbaWidgetTypeStage          = IbaWidgetType{"stage"}
-	IbaWidgetTypeAnomalyHeatmap = IbaWidgetType{"anomaly_heatmap"}
+	IbaWidgetTypeStage          = IbaWidgetType{Value: "stage"}
+	IbaWidgetTypeAnomalyHeatmap = IbaWidgetType{Value: "anomaly_heatmap"}
 	IbaWidgetTypes              = enum.New(IbaWidgetTypeStage, IbaWidgetTypeAnomalyHeatmap)
 )
 
 type IbaWidget struct {
 	Id        ObjectId
-	CreatedAt time.Time
-	UpdatedAt time.Time
+	CreatedAt *time.Time
```

Comment:
Why did you make these pointers as opposed to objects?

Reply:
They became pointers in the raw object so I could send them to the API in the test without having to send a bogus (though likely ignored) "zero time".

I made 'em pointers here in the public object for the same reason, and (I thought) for continuity with other objects. I used LogicalDevice as a template.

...It turns out there's not a ton of consistency here, and what consistency we have seems to point the other direction: Most things don't use a pointer for these time fields. I've just done a quick look around, and found two explanations for that:

1. Some types use a separate "request" object (one each, raw and polished) in their "create" method. This pattern predates the "object with data object" pattern. RackType is an example.
2. Other types seem to just send the "zero time" with every request, and the API conveniently ignores it. It doesn't seem very elegant to always have a time field attached, knowing it may be bogus.

Using a pointer seems like the most "right" way to go in both the raw and polished object, because it's the only way to skip having a value in that element.

Reply:
I tried out the steps in the test.
Basically created_at and upated_at are values set by Apstra upon creation (and presumably) update.
We should never have to fill it out even for testing.
In the event of create/update, we would use a rawWidgetData object.
The logic of our Object/ObjectData breakup was that Object contains everything that Apstra sets and ObjectData has everything that we set.

Let me create a PR on this branch to explain my point a little better.